src/MedTrackerAPI.Tests/FakeFactory.cs

```diff
@@ -0,0 +1,30 @@
+using MedTrackerAPI.Infrastructure;
+
+namespace MedTrackerAPI.Tests;
+
+public static class FakeFactory
+{
+    public static Device CreateFakeDevice()
+    {
+        var autoFaker = new AutoFaker<Device>()
+            .RuleFor(d => d.Id, f => f.Random.Int(min: 1, max: int.MaxValue))
+            .RuleFor(d => d.Supplies, f => []);
+        
+        return autoFaker.Generate();
+    }
+
+    public static Device WithSupplies(this Device device, Action<Supply>? configure = null)
+    {
+        var supplies = new AutoFaker<Supply>()
+            .RuleFor(s => s.Id, f => f.Random.Int(min: 1, max: int.MaxValue))
+            .RuleFor(s => s.DeviceId, f => device.Id)
+            .RuleFor(s => s.Device, f => device)
+            .Generate(new Random().Next(1, 100));
```

Comment:
It’s pretty common to just generate a single one. A pattern you’ll see in tests is to only generate what you need. Generating potentially hundreds of objects for each test, which could be thousands, can cause a burden for test vms like GitHub where you’re riding like 8GB memory. In most cases, you only need a single supply. But if you need two, you can just call WithSupply multiple times and configure each.

Reply:
Makes sense. I updated to just generate one.